core/store.go

```diff
@@ -58,18 +58,42 @@ func NewObj(value interface{}, expDurationMs int64, oType, oEnc uint8) *Obj {
 	return obj
 }
 
-func Put(k string, obj *Obj) {
+type PutOptions struct {
+	KeepTTL bool
+}
+
+func Put(k string, obj *Obj, opts ...PutOption) {
+	options := getDefaultOptions()
+
+	for _, optApplier := range opts {
+		optApplier(options)
+	}
+
```

Comment:
Do you think it will be better if we do this inside the putHelper (and simply pass the opts directly to it)?

That way we can probably get rid of the nil params being passed to putHelper in a few other places. Additionally, we won't need to do an `opts != nil` check inside the helper.

Reply:
Thanks for the note @JyotinderSingh.
I have made the changes for the same.